File: 2023-2024/Gara 2/swappingdigits.py
#!/usr/bin/env python3
# @check-accepted: examples sub1 sub2 full
from sys import stdin, stdout

def rd(): return stdin.readline().strip()
def rdl(x): return map(x, rd().split())
def wt(x): stdout.write(str(x))
def wtl(x): wt(str(x) + '\n')

for qq in range(int(rd())):
  N = rd()
  if len(N) == 1:
    wtl(-1)
    continue
  if N[-2:] in {"00", "25", "50", "75"}:
    wtl(0)
    continue
  if N[-2:] in {"52", "05", "57"}:
    wtl(1)
    continue
  cnt = [0]*10
  for c in N:
    cnt[ord(c) - ord('0')] += 1
  if (N[-1] == '5' and (cnt[2] > 0 or cnt[7] > 0)) or (N[-1] == '0' and (cnt[5] > 0 or cnt[0] > 1)):
    wtl("1")
  elif (N[-2] == '2' and cnt[5] > 0) or (N[-2] == '5' and cnt[0] > 0) or (N[-2] == '7' and cnt[5] > 0) or (N[-2] == '0' and cnt[0] > 1):
    wtl("1")
  elif (cnt[5] > 0 and (cnt[0] > 0 or cnt[2] > 0 or cnt[7] > 0)) or cnt[0] >= 2:
    wtl("2")
  else:
    wtl("-1")

File: 2023-2024/Gara 2/swappingdigits.cpp
// @check-accepted: examples sub1 sub2 full
/**
 *    author:  BERNARD B.01
**/
#include <bits/stdc++.h>

using namespace std;

#ifdef B01
#include "deb.h"
#else
#define deb(...)
#endif

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0);
  int tt;
  cin >> tt;
  while (tt--) {
    string s;
    cin >> s;
    if (s.length() == 1) {
      cout << -1 << '\n';
      continue;
    }
    reverse(s.begin(), s.end());
    string d = s.substr(0, 2);
    if (d == "52" || d == "05" || d == "57" || d == "00") {
      cout << 0 << '\n';
      continue;
    }
    if (d == "25" || d == "50" || d == "75") {
      cout << 1 << '\n';
      continue;
    }
    array<int, 10> cnt = {};
    for (char c : s) {
      cnt[(int) (c - '0')] += 1;
    }
    if ((d[0] == '5' && (cnt[2] > 0 || cnt[7] > 0)) || (d[0] == '0' && (cnt[5] > 0 || cnt[0] > 1))) {
      cout << 1 << '\n';
      continue;
    }
    if ((d[1] == '2' && cnt[5] > 0) || (d[1] == '5' && cnt[0] > 0) || (d[1] == '7' && cnt[5] > 0) || (d[1] == '0' && cnt[0] > 1)) {
      cout << 1 << '\n';
      continue;
    }
    if ((cnt[5] > 0 && (cnt[0] > 0 || cnt[2] > 0 || cnt[7] > 0)) || cnt[0] >= 2) {
      cout << 2 << '\n';
      continue;
    }
    cout << -1 << '\n';
  }
  return 0;
}
// last two digits must be 25, 50, 75, or 00
